imjoy/imjoyPluginEngine.py

```diff
@@ -477,6 +487,10 @@ def killPlugin(pid):
         if "sid" in plugins[pid]:
             if plugins[pid]["sid"] in plugin_sids:
                 del plugin_sids[plugins[pid]["sid"]]
+        if "publish_id" in plugins[pid] and plugins[pid]["publish_id"] is not None:
```

Comment:
This block seems to repeat the new block above. Can we extract it to a helper function?

Reply:
I agree, this file need to be refactored, there are lots of places like that.
I would like to construct classes, and use database to store all these.